plugins/package-managers/cocoapods/src/funTest/kotlin/CocoaPodsFunTest.kt

```diff
@@ -74,5 +78,29 @@ class CocoaPodsFunTest : WordSpec({
 
             result.withInvariantIssues().toYaml() should matchExpectedResult(expectedResultFile, definitionFile)
         }
+
+        "determine dependencies from a Podfile located in a subdirectory referring to npm packages" {
+            // The NPM package manager is run first to initialize the node_modules directory, which is required to
+            // parse the Podfile successfully.
+
+            val npmDefinitionFile = getAssetFile("projects/synthetic/subdirectory/package.json")
+            val result = analyze(
+                npmDefinitionFile.parentFile,
+                true,
+                packageManagers = listOf(NpmFactory(), CocoaPodsFactory()),
+                packageManagerConfiguration = mapOf("CocoaPods" to PackageManagerConfiguration(listOf("NPM")))
+            ).getAnalyzerResult()
+
+            // The NPM-related results are not relevant for this test so they can be discarded.
+            val analyzerResult = result.copy(
+                projects = result.projects.filter { it.id.type == "CocoaPods" }.toSet(),
+                packages = result.packages.filter { it.id.type == "Pod" }.toSet(),
+                issues = emptyMap()
+            )
+            val definitionFile = getAssetFile("projects/synthetic/subdirectory/ios/Podfile")
+            val expectedResultFile = getAssetFile("projects/synthetic/subdirectory-expected-output.yml")
```

Comment:
Line 100, 101 we typically have defined before the `act` phase of the test (arrange, act, asset).